### PR TITLE
Update core to master, add storage back fixtures for Formplayer

### PR DESCRIPTION
**Pull request by wpride**
This merges Formplayer back up with commcare-core master and adds code to enable storage back fixtures for Formplayer as in https://github.com/dimagi/commcare-core/pull/502

### CODE REVIEW

File: src/main/java/session/FormplayerInstanceInitializer.java [hunk 1]
```diff
@@ -29,6 +32,19 @@ public FormplayerInstanceInitializer(FormplayerSessionWrapper formplayerSessionW
...
     @Override
+    protected AbstractTreeElement setupCaseData(ExternalDataInstance instance) {
+        if (casebase == null) {
+            SqliteIndexedStorageUtility<Case> storage = (SqliteIndexedStorageUtility<Case>) mSandbox.getCaseStorage();
+            casebase = new FormplayerCaseInstanceTreeElement(instance.getBase(), storage);
+        } else {
+            //re-use the existing model if it exists.
+            casebase.rebase(instance.getBase());
+        }
+        //instance.setCacheHost((AndroidCaseInstanceTreeElement)casebase);
```
Comment (benrudolph) [inline src/main/java/session/FormplayerInstanceInitializer.java]: comment

File: src/main/java/engine/FormplayerCaseInstanceTreeElement.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * Created by willpride on 1/26/17.
+ */
+public class FormplayerCaseInstanceTreeElement extends CaseInstanceTreeElement {
+
+    public FormplayerCaseInstanceTreeElement(AbstractTreeElement instanceRoot, IStorageUtilityIndexed<Case> storage) {
+        super(instanceRoot, storage);
+    }
+
+    @Override
+    protected Vector<Integer> getNextIndexMatch(Vector<String> keys, Vector<Object> values,
```
Comment (benrudolph) [inline src/main/java/engine/FormplayerCaseInstanceTreeElement.java]: this seems ripe for a test

**Comment by benrudolph** [comment]
if you feel confident in your local QA, feel free to merge

**Comment by wpride** [comment]
QA passed, merging

**Pull request merged**